src/hb-subset-input.cc

```diff
@@ -478,4 +480,26 @@ hb_subset_preprocess (hb_face_t *source)
 
   return new_source;
 }
+
+/**
+ * hb_subset_input_override_name_table:
+ * @input: a #hb_subset_input_t object.
+ * @name_id: name_id of a nameRecord
+ * @name_str: pointer to name string new value or null to indicate should remove
+ *
+ * Override the name string of a nameRecord with specified name_id
+ * Since: EXPERIMENTAL
+ **/
+HB_EXTERN void
+hb_subset_input_override_name_table (hb_subset_input_t  *input,
+                                     unsigned            name_id,
+                                     const char         *name_str)
```

Comment:
Everywhere we take a string we also take its length (where -1 means "use strlen").

Reply:
Also use `hb_ot_name_id_t`

Reply:
Done.